### PR TITLE
Rename MySQL error to use a more generic name

### PR DESCRIPTION
**Pull request by Eragonfr**
## Description

The MysqlError and related types can be reused by other SQL based backends. See PR #1520

## Testing

Test that everything still works like on master

## Issue(s)

Not really an issue but [here's the review](https://github.com/mozilla-services/syncstorage-rs/pull/1520#discussion_r1817098996) about splitting this from #1520

### CODE REVIEW

**Comment by jrconlin** [comment]
For additional context:

eragonfr is working on the sqlite backend for syncstorage. One of the refactoring changes she would like to make is to qualify the `mysql` errors from general `SQL` errors so that she can introduce `sqlite` errors. Since hers will be the third storage engine, the refactor makes sense to me. 

I'd like to get y'all's opinion as well.

File: syncserver-db-common/src/error.rs
Comment (jrconlin) [inline syncserver-db-common/src/error.rs:54]: These metric changes do not impact our dashboards.

**Review by jrconlin — APPROVED** [review]

**Comment by jrconlin** [comment]
I'll poke the other two to get another review before I land this. (Sorry, things have been a bit chaotic over here.)

File: syncserver-db-common/src/error.rs
Comment (pjenvey) [inline syncserver-db-common/src/error.rs:9]: This comment still stands and I think suggests a clearer name for this: `DieselError`, how does that sound?

**Review by taddes — APPROVED** [review]
👍

File: syncserver-db-common/src/error.rs
Comment (jrconlin) [reply syncserver-db-common/src/error.rs:9]: Honestly, I'm fine either way, but I do appreciate the concept of having a more generic `SqlError` for the reason that @Eragonfr notes. I don't really see us replacing diesel any time soon, but the sources of the error being reported are coming from the underlying SQL system. `DieselError` type should be reserved for errors arising from the diesel system directly (e.g. failure to connect to the data store, or some configuration break).

**Pull request merged**
